{
  "schema_version": "1.2.0",
  "id": "GHSA-h4g7-8m7r-87r9",
  "modified": "2022-06-17T22:29:32Z",
  "published": "2022-05-02T04:00:26Z",
  "aliases": [
    "CVE-2009-5012"
  ],
  "summary": "Improper Access Control in pyftpdlib",
  "details": "ftpserver.py in pyftpdlib before 0.5.2 does not require the l permission for the MLST command, which allows remote authenticated users to bypass intended access restrictions and list the root directory via an FTP session.",
  "severity": [

  ],
  "affected": [
    {
      "package": {
        "ecosystem": "PyPI",
        "name": "pyftpdlib"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0.5.2"
            }
          ]
        }
      ],
      "database_specific": {
        "last_known_affected_version_range": "<= 0.5.1"
      }
    }
  ],
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2009-5012"
    },
    {
      "type": "PACKAGE",
      "url": "https://github.com/giampaolo/pyftpdlib"
    },
    {
      "type": "WEB",
      "url": "http://code.google.com/p/pyftpdlib/issues/detail?id=114"
    }
  ],
  "database_specific": {
    "cwe_ids": [
      "CWE-284"
    ],
    "severity": "HIGH",
    "github_reviewed": true
  }
}